# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/events/EventStateManager.cpp
# Commit: 1fa3e699ce84
# Full Hash: 1fa3e699ce84a2c7562e25e2e3a6c1194372042f
# Author: David Parks <davidp99@gmail.com>
# Date: 2024-07-04 16:09:52
# Regressor Bug: 1893119
# File Overlap Count: 1
# Description:
#   Bug 1893119: Part 2 - Make C++ callers of nsIDragService::GetCurrentSession pass a widget r=gstoll,win-reviewers,geckoview-reviewers,m_kato
#   
#   Content process callers must call nsIDragService::GetCurrentSession with the
#   correct widget under the drag.  For all callers, it is a sanity check.
#   
# ==============================================================================

diff -r 21e7f9ac67f8 -r 1fa3e699ce84 dom/events/EventStateManager.cpp
--- a/dom/events/EventStateManager.cpp	Thu Jul 04 07:48:03 2024 +0000
+++ b/dom/events/EventStateManager.cpp	Thu Jul 04 07:48:03 2024 +0000
@@ -2430,8 +2430,8 @@
     nsCOMPtr<nsIDragService> dragService =
         do_GetService("@mozilla.org/widget/dragservice;1");
     if (dragService) {
-      nsCOMPtr<nsIDragSession> dragSession;
-      dragService->GetCurrentSession(getter_AddRefs(dragSession));
+      RefPtr<nsIDragSession> dragSession =
+          dragService->GetCurrentSession(mPresContext->GetRootWidget());
       if (!dragSession) {
         // Only notify if there isn't a drag session active.
         dragService->RemoveAllChildProcesses();
@@ -2823,8 +2823,11 @@
   // began.  However, if we're handling drag session for synthesized events,
   // we need to initialize some information of the session.  Therefore, we
   // need to keep going for synthesized case.
-  nsCOMPtr<nsIDragSession> dragSession;
-  dragService->GetCurrentSession(getter_AddRefs(dragSession));
+  if (MOZ_UNLIKELY(!mPresContext)) {
+    return true;
+  }
+  nsCOMPtr<nsIDragSession> dragSession =
+      dragService->GetCurrentSession(mPresContext->GetRootWidget());
   if (dragSession && !dragSession->IsSynthesizedForTests()) {
     return true;
   }